### DISCUSSION TITLE
Python AI chatbot is not responding to user inputs properly on responses = {}

### DISCUSSION THREAD

**Discussion by andrewetm** [ahmadfaizalbh/Chatbot#82] [Q&A] [answered]
on project 3 ,codecademy AI python chabot building

**Comment by rakib86** [comment, marked as answer]
try this:

import random

# Define a dictionary of responses
responses = {
    "hello": ["Hi there!", "Hello!", "Hey!"],
    "how are you": ["I'm doing well, thanks!", "I'm good. How are you?", "I'm great. How about you?"],
    "what's up": ["Not much. How about you?", "Just chilling. What's up with you?", "Nothing much. You?"],
    "bye": ["Goodbye!", "See you later!", "Bye!"]
}

# Define a function that takes a user input and returns a response
def respond(user_input):
    if user_input in responses:
        return random.choice(responses[user_input])
    else:
        return "I'm sorry, I don't understand what you're saying."

# Define the main function that loops until the user types "bye"
def main():
    print("Hi, I'm a chatbot. What can I help you with?")
    while True:
        user_input = input(">> ")
        if user_input == "bye":
            print(respond(user_input))
            break
        else:
            print(respond(user_input))

# Call the main function to start the chatbot
if __name__ == "__main__":
    main()
